debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java

```diff
@@ -55,9 +56,9 @@
      *
      * @param valueConverterProvider the provider for obtaining {@link ValueConverter}s and {@link SchemaBuilder}s; may not be
      *            null
-     * @param schemaNameValidator the validation function for schema names; may not be null
+     * @param schemaNameValidator the validator for schema names; may not be null
      */
-    public TableSchemaBuilder(ValueConverterProvider valueConverterProvider, Function<String, String> schemaNameValidator, Schema sourceInfoSchema) {
+    public TableSchemaBuilder(ValueConverterProvider valueConverterProvider, AvroValidator schemaNameValidator, Schema sourceInfoSchema) {
```

Comment:
Is it really correct? We are in this case replacing generic with specific? Maybe there should be a parent interface witch validate method that will be extended by AvroValidator. And the parameter should be of that parent interface type.